packages/plugin-preview/src/index.ts

```diff
@@ -60,204 +60,102 @@ export const demoMeta: Record<
-                  return;
-                }
-                const id = generateId(pageName, index++);
-                registerDemo(id, src, isMobileMode);
-              }
-            });
-
-            visit(ast, 'code', (node: Code) => {
-              if (node.lang === 'jsx' || node.lang === 'tsx') {
-                const { value, meta } = node;
-                const hasPureMeta = meta?.includes('pure');
-                const hasPreviewMeta = meta?.includes('preview');
-
-                let noTransform;
-                switch (defaultRenderMode) {
-                  case 'pure':
-                    noTransform = !hasPreviewMeta;
-                    break;
-                  case 'preview':
-                    noTransform = hasPureMeta;
-                    break;
-                  default:
-                    break;
-                }
-
-                // do not anything for pure mode
-                if (noTransform) {
-                  return;
-                }
-
-                // every code block can change their preview mode by meta
-                const isMobileMode =
-                  node?.meta?.includes('mobile') ||
-                  (!node?.meta?.includes('web') && isMobile);
-
-                const { pageName } = routeMeta.find(
-                  meta =>
-                    normalizePosixPath(meta.absolutePath) ===
-                    normalizePosixPath(filepath),
-                )!;
-                const id = generateId(pageName, index++);
-
-                const demoDir = join(
-                  process.cwd(),
-                  'node_modules',
-                  RSPRESS_TEMP_DIR,
-                  `virtual-demo`,
-                );
-
-                const virtualModulePath = join(demoDir, `${id}.tsx`);
-                registerDemo(id, virtualModulePath, isMobileMode);
-
-                fs.ensureDirSync(join(demoDir));
-                fs.writeFileSync(
-                  virtualModulePath,
-                  injectDemoBlockImport(value, demoBlockComponentPath),
-                );
-              }
-            });
-          } catch (e) {
-            console.error(e);
-            throw e;
-          }
-        }),
-      );
-      /**
-       * expect the meta of each demo file is like this:
-       * {
-       *   id,
-       *   component,
-       * }
-       */
-      const virtualMeta = `
-        ${files
-          .map((filepath, index) => {
-            return `import Route${index} from '${filepath}?meta';`;
-          })
-          .join('\n')}
-        export const demos = [${files
-          .map((_, index) => `Route${index}`)
-          .flat()
-          .join(',')}];
-        `;
-      demoRuntimeModule.writeModule('virtual-meta', virtualMeta);
+    },
+    beforeBuild(_, isProduction) {
+      isProd = isProduction;
+    },
+    async afterBuild(config, isProd) {
+      if (isMobile && iframePosition === 'fixed') {
+        // TODO: write entry for all page
```

Comment:
We need write a entry when show device, need preview all demo in one route. like [virtual-demo](https://github.com/web-infra-dev/rspress/blob/main/packages/plugin-preview/src/virtual-demo.tsx#L20-L29)

Reply:
Unfortunately, this cannot be handled at runtime.
We must concatenate strings and manually output them as files.